From 895777890485428a23f2e0da0e8ccd2edf624f6e Mon Sep 17 00:00:00 2001
From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Fri, 25 Jan 2013 03:26:10 +0000
Subject: use mpegaudioparse gstreamer plugin instead of mp3parse

Fix "not suitable for audio or video media" error when
attempting to create a mp3 audio project
The plugin was basically renamed for gstreamer 1.0.
Thanks tuxor for identifying the problem!

https://bugzilla.gnome.org/show_bug.cgi?id=687886
---
diff --git a/libbrasero-utils/brasero-metadata.c b/libbrasero-utils/brasero-metadata.c
index 1088121..7f10088 100644
--- a/libbrasero-utils/brasero-metadata.c
+++ b/libbrasero-utils/brasero-metadata.c
@@ -1200,12 +1200,12 @@ brasero_metadata_create_mp3_pipeline (BraseroMetadata *self)
 	}
 	gst_bin_add (GST_BIN (priv->pipeline_mp3), source);
 
-	parse = gst_element_factory_make ("mp3parse", NULL);
+	parse = gst_element_factory_make ("mpegaudioparse", NULL);
 	if (!parse) {
 		priv->error = g_error_new (BRASERO_UTILS_ERROR,
 					   BRASERO_UTILS_ERROR_GENERAL,
 					   _("%s element could not be created"),
-					   "\"mp3parse\"");
+					   "\"mpegaudioparse\"");
 
 		g_object_unref (priv->pipeline_mp3);
 		priv->pipeline_mp3 = NULL;
diff --git a/plugins/transcode/burn-transcode.c b/plugins/transcode/burn-transcode.c
index a6a0f40..6861168 100644
--- a/plugins/transcode/burn-transcode.c
+++ b/plugins/transcode/burn-transcode.c
@@ -284,13 +284,13 @@ brasero_transcode_create_pipeline_size_mp3 (BraseroTranscode *transcode,
 
 	BRASERO_JOB_LOG (transcode, "Creating specific pipeline for MP3s");
 
-	parse = gst_element_factory_make ("mp3parse", NULL);
+	parse = gst_element_factory_make ("mpegaudioparse", NULL);
 	if (!parse) {
 		g_set_error (error,
 			     BRASERO_BURN_ERROR,
 			     BRASERO_BURN_ERROR_GENERAL,
 			     _("%s element could not be created"),
-			     "\"Mp3parse\"");
+			     "\"mpegaudioparse\"");
 		g_object_unref (pipeline);
 		return FALSE;
 	}
--
cgit v0.9.1
